File: boat_app_client/src/app/class/boatdetails.ts

export class BoatDetails {
 
    id: number=0;
    boatIdfk: number=0;
    weight: number=0;
    manufacturedDate: Date;
    quantity: number=0;
    type: string="";

    constructor(){
        this.manufacturedDate = new Date(Date.now());
    }

}


export const TestBoatDetails=[
    {id:1, boatIdfk:"1", weight: "100.5", manufacturedDate:"10.10.1990", quantity:"3", type: "type A"},
    {id:2, boatIdfk:"2", weight: "202", manufacturedDate:"02.10.1885", quantity:"1", type: "type B"},
    {id:3, boatIdfk:"4", weight: "150.5", manufacturedDate:"06.05.2000", quantity:"5", type: "type D"},
]

File: boat_app/src/main/java/ch/openwt/shop/boat_app/model/BoatDetails.java
package ch.openwt.shop.boat_app.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
@NoArgsConstructor
public class BoatDetails {

    private Integer id;
    private Integer boatIdfk;
    private Double weight;
    private Date manufacturedDate;
    private Integer quantity;
    private String type;

}
